### PR TITLE
Support Twig 2 & cleaning up the Twig extension

### PR DESCRIPTION
**Pull request by WouterJ**

### CODE REVIEW

File: Twig/Extension/SonataDoctrinePHPCRAdminExtension.php [hunk 1]
```diff
@@ -17,55 +17,13 @@
...
-    /**
-     * render a list element from the FieldDescription.
-     *
-     * @param object                    $object
-     * @param FieldDescriptionInterface $fieldDescription
-     * @param array                     $params
-     *
-     * @return string
-     */
-    public function renderListElement($object, FieldDescriptionInterface $fieldDescription, $params = array())
```
Comment (WouterJ) [inline Twig/Extension/SonataDoctrinePHPCRAdminExtension.php]: this was added in https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/commit/638ee10b84976650a7b35777c024dcd185aeb742 but it isn't called from anywhere. To me, this seems like dead code that should be removed.

File: Twig/Extension/SonataDoctrinePHPCRAdminExtension.php [hunk 2]
```diff
@@ -17,55 +17,13 @@
 class SonataDoctrinePHPCRAdminExtension extends \Twig_Extension
 {
     /**
-     * @var \Twig_Environment
-     */
-    protected $environment;
```
Comment (WouterJ) [inline Twig/Extension/SonataDoctrinePHPCRAdminExtension.php]: The environment property is never used

File: Twig/Extension/SonataDoctrinePHPCRAdminExtension.php [hunk 3]
```diff
@@ -17,55 +17,13 @@
...
-     */
-    public function initRuntime(\Twig_Environment $environment)
-    {
-        $this->environment = $environment;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getName()
```
Comment (WouterJ) [inline Twig/Extension/SonataDoctrinePHPCRAdminExtension.php]: moved down, as that's the common order in all other Twig extension.

**Comment by lsmith77** [comment]
maybe better to move to styleci?

```
1) Twig/Extension/SonataDoctrinePHPCRAdminExtension.php (unused_use)
```

**Comment by OskarStark** [comment]
@lsmith77 @Soullivaneuh wants to use StyleCI, maybe he can test his travis integration in this bundle.

or he already did, and i don't know.

**Comment by Soullivaneuh** [comment]
@lsmith77 please see and support https://github.com/sonata-project/SonataAdminBundle/pull/3273

If it's accepted, I'll make it available on each sonata project.

**Comment by dbu** [comment]
i propose we do the php version => 5.5 change either here or in #348 and then rebase the other branch. wdyt @WouterJ ? which branch is simpler to wrap up? or should we do that change in a separate PR and rebase both branches?

**Comment by WouterJ** [comment]
Fixed CS and bumped the mim PHP version to 5.5. Ready to be merged imo

**Comment by dbu** [comment]
there seems to be a CS issue https://travis-ci.org/sonata-project/SonataDoctrinePhpcrAdminBundle/jobs/87634558

**Comment by WouterJ** [comment]
@dbu I think that's a bug as it tries to align the second `=>`.

File: composer.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
         }
     ],
     "require": {
-        "php": ">=5.3.9",
+        "php": ">=5.5",
```
Comment (Soullivaneuh) [inline composer.json]: Why?

**Comment by OskarStark** [comment]
yes, you can ignore this, it's related to: FriendsOfPHP/PHP-CS-Fixer#1450

its fixed, but not yet released

**Comment by dbu** [comment]
thanks wouter!

**Pull request merged**